modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -152,6 +152,41 @@ object CrossSources {
         .map(_.flatten)
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
+    val preprocessedWithUsingDirs = preprocessedInputFromArgs.filter(_.directivesLocation.isDefined)
```

Comment:
Do you test your mechanism with `file` directives?
```
//> using file "Utils.scala"
```

Reply:
No I do not. I will change that.